aws/rust-runtime/aws-config/src/connector.rs

```diff
@@ -51,7 +51,24 @@ pub fn default_connector(
 }
 
 /// Given `ConnectorSettings` and an `AsyncSleep`, create a `DynConnector` from defaults depending on what cargo features are activated.
-#[cfg(not(any(feature = "rustls", feature = "native-tls")))]
+#[cfg(all(
+    target_family = "wasm",
+    target_os = "wasi",
+    not(any(feature = "rustls", feature = "native-tls"))
+))]
+pub fn default_connector(
+    _settings: &ConnectorSettings,
+    _sleep: Option<Arc<dyn AsyncSleep>>,
+) -> Option<DynConnector> {
+    Some(DynConnector::new(crate::wasi_adapter::Adapter::default()))
+}
+
+/// Given `ConnectorSettings` and an `AsyncSleep`, create a `DynConnector` from defaults depending on what cargo features are activated.
+#[cfg(not(any(
+    all(target_family = "wasm", target_os = "wasi"),
+    feature = "rustls",
+    feature = "native-tls"
+)))]
```

Comment:
how do you feel about making a separate crate that has this connector in it? I don't think it really belongs in aws-config—for one thing, folks that are using smithy-rs directly and not using any AWS features might want to use it. If we want, we _could_ consider adding a feature to `aws-config` to opt into turning it on, but I think the code itself should be isolated. This is where we can also eventually put the WASM time source, e.g.

Reply:
I completely agree! Should I name the crate `aws-smithy-wasm`?